app/controllers/session_controller.rb

```diff
@@ -0,0 +1,45 @@
+
+class SessionController < ApplicationController
+
+  def authenticate
+    config = OpenIDConnect::Discovery::Provider::Config.discover! "https://did.app"
+
+    client = OpenIDConnect::Client.new({
```

Comment:
Work out where to put the client and config.
Is the correct place as an initializer? Or is it somewhere in boot. 
The `discover!` function makes a webcall but there would be no harm hardcoding the authorization_endpoint etc.

Reply:
I think we can put it in an ENV variable still, no?

Reply:
That doesn't make sense to me.
I mentioned the `discover!` call makes a web request. and should be done in some startup step

Reply:
Hey Peter — what's your specific question here? I can't tell if you're asking me something or telling me something. 😅

Reply:
I am asking, where do I put code that makes a web request during startup?

"https://did.app" can go in the env variable. But then there is a call that should be performed only once to lookup the authorization endpoint etc. that is what discover does. The `config = ` line should not be part of the http handler but some boot or startup code

Reply:
Ah, got it!

Well, I think you want a new file under `config/initializers`.  See more here: https://guides.rubyonrails.org/configuring.html#using-initializer-files.